internal/grpcserver/storage/elastic.go

```diff
@@ -0,0 +1,213 @@
+package storage
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"github.com/astanishevskyi/grpc-server/internal/grpcserver/models"
+	elastic "github.com/elastic/go-elasticsearch/v7"
+	"github.com/elastic/go-elasticsearch/v7/esapi"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+type ElasticStorage struct {
+	client *elastic.Client
+	mu     sync.Mutex
+	lastID uint32
+}
+
+func NewElasticStorage() *ElasticStorage {
+	user := os.Getenv("ELASTIC_USER")
+	pass := os.Getenv("ELASTIC_PASS")
+	addr := os.Getenv("ELASTIC_ADDR")
+
+	client, err := elastic.NewClient(elastic.Config{
+		Addresses: []string{addr},
+		Username:  user,
+		Password:  pass,
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+	if _, err := client.Ping(); err != nil {
+		log.Fatal(err)
+	}
+	e := &ElasticStorage{client: client}
+	lastID, _ := e.getLastID()
+	e.lastID = lastID
+	return e
+}
+
+func (e *ElasticStorage) getLastID() (uint32, error) {
+	var buf bytes.Buffer
+	query := map[string]interface{}{
```

Comment:
if the query is static, why do you encode it on each call. Just store `[]byte` from string somewhere in constant in order to reduce the memory allocation